### PR TITLE
fix(npm): improve package suggestions

### LINKED ISSUE #275: [yarn/npm] yarn/npm package searches only suggest with full word results
I'm not sure if this is intended to reduce the number request of but it's confusing :confused:. Suggestion to make it show suggestions with any character typed

https://user-images.githubusercontent.com/4255482/121294170-25c48900-c917-11eb-84a5-3cbf19f1cfd9.mov

### LINKED ISSUE #974: [npm] rank popular packages higher
### Sanity checks

- [X] My issue relates to a specific CLI completion spec (e.g. `git checkout` is missing options in `git` completion spec). If your issue is more general, please create your issue here: [withfig/fig](https://github.com/withfig/fig/issues/new?assignees=&labels=NEED_TO_LABEL&template=main_issue_template.yml)
- [X] I have searched [github.com/withfig/autocomplete/issues](https://github.com/withfig/autocomplete/issues?q=) and there are no duplicates of my issue
- [ ] I would like to work on this.

### What CLI tool does this relate to?

npm

### Which statement makes the most sense?

Something else

### Issue Details

Currently if I write `npm i vi` I get this:
CleanShot 2022-02-25 at 13 02 49@2x

I think most people would expect to see `vite` near the top given how popular it is.

### PR DESCRIPTION
**Pull request by loosheng**
**What kind of change does this PR introduce? (Bug fix, feature, docs update, ...)**
pref 
**What is the current behavior? (You can also link to an open issue here)**
close #275
close #974
**What is the new behavior (if this is a feature change)?**

better-npm

**Additional info:**
use https://api.npms.io/v2/search/suggestions?q=%q&size=20

### CODE REVIEW

File: src/npm.ts [hunk 1]
```diff
@@ -25,18 +25,23 @@ export const createNpmSearchHandler =
...
-
     // Add optional keyword parameter
     const keywordParameter =
       keywords?.length > 0 ? `+keywords:${keywords.join(",")}` : "";
+
+    let queryPackagesUrl = `https://api.npms.io/v2/search/suggestions?q=${searchTerm}&size=20`;
+
+    if (keywordParameter) {
+      queryPackagesUrl = `https://api.npms.io/v2/search?size=20&q=${searchTerm}${keywordParameter}`;
+    }
```
Comment (fedeci) [inline src/npm.ts:36]:
```suggestion
    const queryPackagesUrl = keywordParameter
      ? `https://api.npms.io/v2/search/suggestions?q=${searchTerm}&size=20`
      : `https://api.npms.io/v2/search?size=20&q=${searchTerm}${keywordParameter}`;
```

File: src/npm.ts [hunk 2]
```diff
@@ -97,7 +103,7 @@ export const npmSearchGenerator: Fig.Generator = {
   },
   getQueryTerm: "@",
   cache: {
-    ttl: 1000 * 60 * 60 * 24 * 2, // 2 days
+    ttl: 1, // 2 days
```
Comment (fedeci) [inline src/npm.ts:106]: This is 1ms

Comment (loosheng) [reply src/npm.ts]: done.
failure, legacy test code. 😂

**Review by fedeci — APPROVED** [review]
Thanks!

**Comment by fedeci** [comment]
recheck

**Comment by loosheng** [comment]
I have read the CLA Document and I hereby sign the CLA

**Pull request merged**